test/cli_test.py

```diff
@@ -219,6 +224,68 @@ def test_unkown_collection(self):
         out = self.runcli('alt', 'update', 'unkown')
         self.assertIn("Alternative collection 'unkown' not found.", out)
 
+    def test_embed_art(self):
+        """ Test that artwork is embedded and updated to match the source file.
+
+        There used to be a bug that meant that albumart was only embedded
+        once on initial addition to the alternative collection, but not if
+        the artwork was added or changed later.
+
+        This test comprehensively checks that embedded artwork is up-to-date
+        with the artwork file, even if no changes to the database happen.
+        """
+        # Initially add album without artwork.
+        album = self.add_album(myexternal='true')
+        album.store()
+        self.runcli('alt', 'update', 'myexternal')
+
+        item = album.items().get()
+        self.assertHasNoEmbeddedArtwork(item['alt.myexternal'])
+
+        # Make a copy of the artwork, so that changing mtime/content won't
+        # affect the repository.
+        def rm_art():
+            try:
+                os.remove(fname)
+            except OSError:
+                # If the test succeeds, the file will have been removed already
+                pass
+
+        fh, fname = tempfile.mkstemp()
+        os.close(fh)
+        self.addCleanup(rm_art)
```

Comment:
You can use `mkdtemp` from `TestHelper`. It will take care of cleaning up the directory on teardown.
~~~python
image_dir = self.mkdtemp()
image_path = os.path.join(image_dir, 'image')
shutil.copy(self.IMAGE_FIXTURE1, image_path)
~~~